src/audio/google/google_ctc_audio_processing.c

```diff
@@ -0,0 +1,360 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2024 Google LLC.
+//
+// Author: Eddy Hsu <eddyhsu@google.com>
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/format.h>
+#include <sof/audio/pipeline.h>
+
+#include <google_ctc_audio_processing.h>
+
+LOG_MODULE_REGISTER(google_ctc_audio_processing, CONFIG_SOF_LOG_LEVEL);
+
+/* bf0e1bbc-dc6a-45fe-bc90-2554cb137ab4 */
+DECLARE_SOF_RT_UUID("google-ctc-audio-processing", google_ctc_audio_processing_uuid,
+		    0xbf0e1bbc, 0xdc6a, 0x45fe, 0xbc, 0x90, 0x25, 0x54, 0xcb,
+		    0x13, 0x7a, 0xb4);
+
+DECLARE_TR_CTX(google_ctc_audio_processing_tr, SOF_UUID(google_ctc_audio_processing_uuid),
+	       LOG_LEVEL_INFO);
+
+struct google_ctc_audio_processing_comp_data {
+	struct comp_buffer *input;
+	struct comp_buffer *output;
+	uint32_t num_frames;
+	GoogleCtcAudioProcessingState *state;
+	struct comp_data_blob_handler *tuning_handler;
+	bool reconfigure;
+};
+
+static int ctc_free(struct processing_module *mod)
+{
+	struct google_ctc_audio_processing_comp_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
+
+	if (cd) {
+		rfree(cd->input);
+		rfree(cd->output);
+		if (cd->state) {
+			GoogleCtcAudioProcessingFree(cd->state);
+			cd->state = NULL;
+		}
+		rfree(cd);
+	}
+	rfree(dev);
```

Comment:
The processing modules do not need to free mod->dev. It's done in module_adapter_free() in module_adapter.c.

Reply:
@eddy1021 Also please check the codestyle / checkpatch results from CI (long lines, white space).

Reply:
Thanks! I've fixed the long lines and white space issues.